### DISCUSSION TITLE
.NET Core console app references .NET Framework assembly

### DISCUSSION THREAD

**Discussion by avivanoff** [dotnet/runtime#48705] [Q&A]
Sorry for the question, I must be missing something, or doing something completely wrong.

I just created a .NET Core 3.1 console app that references a .NET Framework 4.8 assembly. The console app calls some methods in the referenced assembly. My question is: how does it compile and run?

**Comment by huoyaoyuan** [comment]
There is a shim of `mscorlib.dll` in .NET Core runtime package, which delegates all the supported types to `System.Private.CoreLib`. When a type from mscorlib is attempted to be loaded, the type loader sees`TypeForwardedTo` on the shim and loads from .NET Core assembly.

**Reply by avivanoff** [reply]
Is there a detailed documentation of supported types/scenarios? Or, more importantly, unsupported ones?

**Reply by PathogenDavid** [reply]
Note that while it _works_, it is not a good idea or supported. (If you rebuild your app you should see a build warning.)

If you need to share code between .NET Core and .NET Framework, you should use .NET Standard 2.0.

**Reply by huoyaoyuan** [reply]
I don't know a direct documentation for this, but it should be easy to infer.

Use [API browser](https://docs.microsoft.com/en-us/dotnet/api/) to find the type. If it "exists" in certain .NET Core version, the infrastructure would get it loaded correctly. If it's marked as in platform extension, the corresponding nuget package should be required. Installing [Microsoft.Windows.Compatibility](https://www.nuget.org/packages/Microsoft.Windows.Compatibility/) should be enough for most cases.

In some cases, a type exists, but corresponding feature is unsupported and throws `PlatformUnsupportedException`. You should read the documentation of the type to find out this.
